utils/docker/Dockerfile.ubuntu

```diff
@@ -10,24 +11,22 @@
 ARG BASE_DISTRO=ubuntu:22.04
 FROM $BASE_DISTRO
 LABEL maintainer="daos@daos.groups.io"
+# Needed for later use of BASE_DISTRO
+ARG BASE_DISTRO
 
 # Intermittent cache-bust.  Used to reduce load on the actual CB1 later.
 ARG CB0
 
 ENV DEBIAN_FRONTEND=noninteractive
 
 # Install basic tools
-COPY ./utils/scripts/install-ubuntu.sh /tmp/install.sh
-RUN echo "APT::Get::Assume-Yes \"true\";" > /etc/apt/apt.conf.d/no-prompt
-RUN echo "APT::Install-Recommends \"false\";" > /etc/apt/apt.conf.d/no-recommends
-RUN apt-get update && \
-    apt-get upgrade && \
-    apt-get install gpg-agent software-properties-common && \
-    add-apt-repository ppa:longsleep/golang-backports && \
-    apt-get update && \
-    chmod +x /tmp/install.sh && \
-    /tmp/install.sh && \
-    apt-get clean all
+ARG DAOS_LAB_CA_FILE_URL
+# script to setup local repo if available
+COPY ./utils/scripts/helpers/repo-helper-ubuntu.sh /tmp/repo-helper.sh
+
+RUN chmod +x /tmp/repo-helper.sh &&                 \
+    /tmp/repo-helper.sh &&                          \
+    rm -f /tmp/repo-helper.sh
```

Comment:
Curious why we can't just run this directly?

Reply:
Getting rid of duplicated code in jenkinsfiles in favor of common scripts.

Long chains of linux commands are hard to read, and hard to debug in dockerfiles and in strings in groovy code, so should be avoided.

The old logic was also incomplete, and the new logic is better and works for some Ubuntu packaging, but a lot more work will be needed for Ubuntu.   This is just a small step in the process.

Reply:
Wouldn't it be less duplicated code to just do this?
```
RUN ./utils/scripts/helpers/repo-helper-ubuntu.sh
```
Or maybe I misunderstand why this `CP` is needed

Reply:
Dockerfile RUN can only run scripts that are already in the container at that step.
Dockerfile COPY moves a file from the host system into the container.   You need to make a script executable after the copy.

Reply:
> RUN can only run scripts that are already in the container at that step

Thanks - I'm not super familiar with docker